### PR TITLE
Modifications to imported a11y base classes to build in Flutter

### PR DESCRIPTION
**Pull request by gw280**
This is the full set of modifications needed to get the following `base/` classes to build in Flutter. Work done by both me and @cbracken.

### CODE REVIEW

**Comment by gw280** [comment]
So this isn't yet wired up to Flutter on this branch. I think the best course forwards is to wire up the unittests that were imported and build those.

File: third_party/accessibility/gfx/range/range.cc
Comment (chunhtai) [inline third_party/accessibility/gfx/range/range.cc:15]: we have StringPrintf in base/string_utils.h

File: third_party/accessibility/gfx/range/range_unittest.cc
Comment (chunhtai) [inline third_party/accessibility/gfx/range/range_unittest.cc:15]: We don't need RangeF? I thought they work side by side.

**Review by chunhtai — COMMENTED** [review]
Mostly looks good.

File: third_party/accessibility/gfx/range/range_unittest.cc
Comment (gw280) [reply third_party/accessibility/gfx/range/range_unittest.cc:15]: Correct, we don't need `RangeF`. If we need it in future we can resurrect these unit tests but it's not needed for a11y on Windows.

**Review by cbracken — APPROVED** [review]
rslgtm. I didn't go over this in detail since we did a first pass together when we paired on a bunch of it.

File: third_party/accessibility/BUILD.gn
Comment (cbracken) [inline third_party/accessibility/BUILD.gn:100]: This needs to be in both blocks or in an `if (is_mac || is_win) {` block to avoid making other platforms sad.

Comment (gw280) [reply third_party/accessibility/BUILD.gn]: Done

**Review by cbracken — CHANGES_REQUESTED** [review]
Before landing, can you rebase this into two (or three) commits:
1. Initial import of.*
2. Reformat
3. Patches to get it working with Flutter

If you _have_ items 2 and 3 in separate commits right now, then restacking as three commits is ideal, otherwise it's totally fine to combine them into a single item 2.

**Comment by gw280** [comment]
The initial import patches have been split off into https://github.com/flutter/engine/pull/25057, so once that's landed and this branch is rebased this can be squashed into a single commit.

File: third_party/accessibility/gfx/range/range.cc
Comment (chunhtai) [reply third_party/accessibility/gfx/range/range.cc:15]: This is not addressed yet

Comment (gw280) [reply third_party/accessibility/gfx/range/range.cc]: Thanks for the reminder, fixed now

**Review by chunhtai — COMMENTED** [review]
LGTM, if @cbracken also approves

**Comment by gw280** [comment]
CLAs are good as we're both Googlers

**Review by cbracken — APPROVED** [review]
lgtm

**Pull request merged**
